### DISCUSSION TITLE
Pan left/right reversed?

### DISCUSSION THREAD

**Discussion by thatmichaelpark** [probonopd/MiniDexed#285] [Q&A]
Hi, I have MiniDexed running on an RPi 4 with [this PCM5102A audio interface](https://smile.amazon.com/gp/product/B07Q9K5MT8/ref=ppx_yo_dt_b_asin_title_o02_s00?ie=UTF8&psc=1). It sounds good but panning sounds to the left (as shown on the LCD) actually pans them to the right (in my headphones) and vice versa. Has anyone else encountered this?

**Comment by Banana71** [comment]
Hi Micheal,
I haven't tried it myself, but you can set ChannelsSwapped=1 in minidexed.ini
I hope this solves the problem
Peter

**Reply by thatmichaelpark** [reply]
Thanks for pointing that out, Peter. Unfortunately, setting it to 1 had no effect :(

It should have worked!

**Reply by probonopd** [reply]
Hi @thatmichaelpark. Can you please try `ChannelsSwapped=0` and then `ChannelsSwapped=1`. Do you get any difference? Which sound device are you using, and which Raspberry Pi device?

**Reply by thatmichaelpark** [reply]
No difference between `ChannelsSwapped=0` and `ChannelsSwapped=1`: sounds that are supposed to be panned hard left (right) play in headphones hard right (left).

 Just to convince myself that I was actually editing minidexed.ini, I also tried setting `LCDEnabled` to 1 and 0 and was able to verify that the LCD was enabled or not.

Raspberry Pi 4 and PCM5012A DAC (link in first post).

**Comment by Banana71** [comment]
I have experimented a bit with the components that are available to me with the following result:

- Raspberry PI 3 Model B with DAC raspiaudi Audio+  V2 (miniDexed Version from 03 June 22)
  Pan left & right are in the correct position, ChannelsSwapped=0 or 1 in minidexed.ini has no effect
![raspiaudi_audioV2plus](https://user-images.githubusercontent.com/104296205/174761170-6c1b0c1f-e3e3-4a25-a197-fb7346e63023.jpg)

- Raspberry PI 3 Model B+ with DAC GY-PCM5102 I2S (miniDexed Version from 15 June 22)
  Pan left & right are in the **wrong** position, ChannelsSwapped=0 or 1 in minidexed.ini has no effect
![GY_PCM5102](https://user-images.githubusercontent.com/104296205/174761192-a1ebb217-893a-4bbc-b689-7b3d869d9d08.jpg)

Both DACs use the PCM5102 chip.

**Reply by thatmichaelpark** [reply]
Thanks for testing, Peter. My DAC looks like the one in your second picture. I'm somewhat relieved to hear that you and I are in the same boat :)

**Reply by Banana71** [reply]
a sorrow shared is a sorrow halved :-)
I use a mixer and can cross the cables but i hope someone will find a solution in the code.

**Reply by probonopd** [reply]
What happens if you use pwm (headphone jack) or hdmi for audio, does `ChannelsSwapped` have an effect then?

**Reply by Banana71** [reply]
It's the same with PWM and HDMI, ChannelsSwapped has no effect.

**Reply by ramphands** [reply]
I'm also in the same boat - ChannelsSwapped has no effect.  Using the same board as the ones shown above.

**Reply by mongio** [reply]
Same here, left and right channels inverted with the usual purple dac :(. ChannelsSwapped didn’t work.

**Reply by soyersoyer** [reply]
https://github.com/probonopd/MiniDexed/pull/953
